search/includes/classes/class-health.php

```diff
@@ -460,6 +460,28 @@ public static function diff_document_and_prepared_document( $document, $prepared
 		return $diff;
 	}
 
+	/**
+	 * Iterate over an array of inconsistencies and address accordingly.
+	 *
+	 * If an object is missing from the index - add it to the queue for the sweep.
+	 *
+	 * If an object is missing from the DB, remove it from the index.
+	 *
+	 * @param array $diff array of inconsistenices in the following shape: [ id => string, type => string (Indexable), issue => <missing_from_index|extra_in_index> ].
+	 */
+	public static function reconcile_diff( array $diff ) {
+		foreach ( $diff as $key => $obj_to_reconcile ) {
+			switch ( $obj_to_reconcile['issue'] ) {
+				case 'missing_from_index':
+					\Automattic\VIP\Search\Search::instance()->queue->queue_object( $obj_to_reconcile['id'], $obj_to_reconcile['type'] );
+					break;
+				case 'extra_in_index':
```

Comment:
@nickdaugherty @brandon-m-skinner did I miss the "inconsistent" case, which may happen when `--inspect` is passed.

Reply:
@rinatkhaziev I think so: https://github.com/Automattic/vip-go-mu-plugins/blob/ac59632bb576b7962fc27e2cfaa46342c0393d3c/search/includes/classes/class-health.php#L319-L333

Inconsistent should be treated like missing_from_index since it just needs a reindex to be fixed.